src/shackle_server.erl

```diff
@@ -110,19 +112,19 @@ handle_msg({Request, #cast {
         queue = Queue,
         socket = Socket
     } = State, ClientState}) ->
-
+    shackle_events:queued_time(Client, duration_since(Timestamp)),
     try Client:handle_request(Request, ClientState) of
         {ok, ExtRequestId, Data, ClientState2} ->
             case Protocol:send(Socket, Data) of
                 ok ->
-                    shackle_telemetry:send(Client, iolist_size(Data)),
+                    shackle_events:send(Client, iolist_size(Data)),
```

Comment:
In either case, it is not a good idea to calculate the Data size here, until we know we will use it. It is better to use macro in general and pass Data to telemetry/events function to let it calc size only if needed (if dynamic handlers exist).

Reply:
Great point!

I introduced a macro in shackle_events that will avoid computing these values and terms if there is no configured shackle_hooks handler.